File: app/src/main/java/com/gals/prayertimes/Menu.java
package com.gals.prayertimes;

import android.content.SharedPreferences;
import android.net.Uri;
import android.os.Bundle;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import android.view.MenuItem;

// in order to get the Fragments to work you need to implement those Interactions interfaces it is necessary.
public class Menu extends AppCompatActivity implements AthanSettingsFragment.OnFragmentInteractionListener, MenuItemFragment.OnListFragmentInteractionListener, PrivacyPolicyFragment.OnFragmentInteractionListener {

    public static final String PREFS_NAME = "MyLocalStorage";
    ToolsManager tools;
    SharedPreferences settings;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_menu);

        tools = new ToolsManager(getApplicationContext());

        ActionBar actionBar = getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);
        actionBar.setIcon(R.drawable.mysalay);

        MenuItemFragment menuSettings = new MenuItemFragment();
        getSupportFragmentManager().beginTransaction().replace(R.id.fragmentContainer, menuSettings).commit();

        tools.setActivityLanguage("en");
        tools.changeStatusBarColor(getWindow());
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        //Back to main UI
        switch (item.getItemId()) {
            case android.R.id.home:
                onBackPressed();
                break;
            default:
                break;
        }
        return true;
    }

    @Override
    public void onFragmentInteraction(Uri uri) {

    }


    @Override
    public void onListFragmentInteraction(SettingsMenuContent.SettingsMenuItem item) {

    }
}

File: app/src/main/java/com/gals/prayertimes/ui/Menu.kt
package com.gals.prayertimes.ui

import android.net.Uri
import android.os.Bundle
import android.view.MenuItem
import androidx.appcompat.app.ActionBar
import androidx.appcompat.app.AppCompatActivity
import com.gals.prayertimes.R
import com.gals.prayertimes.ui.MenuItemFragment.OnListFragmentInteractionListener
import com.gals.prayertimes.ui.SettingsMenuContent.SettingsMenuItem
import com.gals.prayertimes.utils.UtilsManager

class Menu : AppCompatActivity(), AthanSettingsFragment.OnFragmentInteractionListener, OnListFragmentInteractionListener, PrivacyPolicyFragment.OnFragmentInteractionListener {
    var tools: UtilsManager? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_menu)
        tools = UtilsManager(applicationContext)
        val actionBar: ActionBar? = supportActionBar
        actionBar!!.setDisplayHomeAsUpEnabled(true)
        actionBar.setIcon(R.drawable.mysalay)
        val menuSettings = MenuItemFragment()
        supportFragmentManager.beginTransaction().replace(
            R.id.fragmentContainer,
            menuSettings
        ).commit()
        tools!!.setActivityLanguage("en")
        tools!!.changeStatusBarColor(window)
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        /**Back to main UI*/
        when (item.itemId) {
            android.R.id.home -> onBackPressed()
            else -> {}
        }
        return true
    }

    override fun onFragmentInteraction(uri: Uri?) {}
    override fun onListFragmentInteraction(item: SettingsMenuItem?) {}
}